### ISSUE TITLE
Enhancer classification: AlphaGenome convolutional probe

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/bolinas-dna#99]
## Goal

Train a binary enhancer classifier using convolutional features from [AlphaGenome PyTorch](https://github.com/genomicsxai/alphagenome-pytorch). The classifier takes train and validation parquet splits as input (produced by #98) and is agnostic to how those splits were defined.

## Model architecture

- **Backbone**: Convolutional trunk from AlphaGenome (`alphagenome-pytorch` on PyPI)
  - Extract only the convolutional/short-range component (before transformer/attention layers)
  - Need to inspect the source to identify the exact module — likely the downsampling stages of the U-Net backbone
  - Input: one-hot encoded 255bp DNA sequences (4 channels: ACGT, N → all zeros)
  - The conv trunk is likely shared across organisms (organism-specific conditioning may only apply at later stages) — verify during implementation
- **Head**: global average pool over sequence dimension → linear layer or 2-layer MLP with ReLU
- **Freezing**: configurable — support both frozen backbone (linear probe) and unfrozen (fine-tuning)

## Training setup

- Framework: PyTorch Lightning
- Loss: BCE with logits
- Optimizer: Adam
- Metrics: AUROC, AUPRC (logged per epoch)
- Early stopping on validation loss
- Model configurations should be parameterized (frozen vs unfrozen, learning rate, batch size, max epochs, head type, seed, etc.)

## Evaluation

- Run on any provided evaluation split (validation, cross-species, etc.)
- Output: JSON with AUROC, AUPRC, accuracy at optimal threshold

## Dependencies

Add a new dependency group to `pyproject.toml`:

```toml
[dependency-groups]
enhancer-classification = [
    "lightning>=2.0",
    "alphagenome-pytorch",
]
```

## Code structure

Model, dataset, and training logic go in `src/bolinas/` (with optional dependency on the `enhancer-classification` group). Snakemake rules call thin entry points.

```
src/bolinas/enhancer_classification/
├── __init__.py
├── dataset.py       # EnhancerDataset (loads parquet, one-hot encodes)
├── model.py         # EnhancerClassifier (LightningModule)
├── train.py         # training entry point
└── evaluate.py      # evaluation entry point

snakemake/enhancer_classification/
├── workflow/
│   ├── rules/
│   │   └── model.smk
```

## Implementation notes

- Build the training loop ourselves (not using AlphaGenome's built-in `TransferConfig`), since we only want the conv trunk. Can reference their `scripts/finetune.py` for how the architecture is structured.
- Input data: parquet files produced by #98 with `genome`, `chrom`, `start`, `end`, `strand`, `seq`, `label` columns
